Components/LearningCore/Source/gov/sandia/cognition/learning/algorithm/clustering/SNNClusterer.java

```diff
@@ -0,0 +1,637 @@
+                if (sharedNeighbors.size() >= this.eps)
+                {
+                    result.add(otherPoint);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Takes a core point and adds its neighbors to the cluster. Also merges any
+     * neighboring clusters by iterating through neighbors and seeing if they
+     * are core points. Records any processed points as "clustered".
+     *
+     * @param point The core point to expand the cluster around
+     * @param neighbors The core point's neighbors.
+     */
+    private void expandCluster(DataType point,
+        List<DataType> neighbors)
+    {
+        LinkedList<DataType> neighborsLinkedList = new LinkedList<>(neighbors);
+        this.currentCluster.add(point);
+        this.clustered.add(point);
+
+        // Loop through each neighbor and process it
+        while (!neighborsLinkedList.isEmpty())
+        {
+            DataType neighbor = neighborsLinkedList.remove();
+
+            if (!this.visited.contains(neighbor))
+            {
+                this.visited.add(neighbor);
+                List<DataType> neighborNeighbors = this.getConnectedNeighbors(
+                    neighbor);
+
+                // If neighbor is a core point, process its neighbors too
+                if (neighborNeighbors.size() >= this.minPts)
+                {
+                    neighborsLinkedList.addAll(neighborNeighbors);
+                }
+            }
+            // Mark neighbors as clustered and add them to current cluster
+            if (!this.clustered.contains(neighbor))
+            {
+                this.currentCluster.add(neighbor);
+                this.clustered.add(neighbor);
+            }
+        }
+    }
+
+    /**
+     * Adds noise cluster to list of clusters.
+     */
+    @Override
+    protected void cleanupAlgorithm()
+    {
+        this.clusters.set(0, this.creator.createCluster(this.noiseCluster));
+    }
+
+    /**
+     * Gets a list of the clusters.
+     *
+     * @return a list of the clusters
+     */
+    @Override
+    public Collection<ClusterType> getResult()
+    {
+        return this.getClusters();
+    }
+
+    /**
+     * Sets the number of nearest neighbors to use.
+     *
+     * @param k The number of nearest neighbors k.
+     */
+    public void setNumNeighbors(int k)
+    {
+        this.k = k;
+    }
+
+    /**
+     * Gets the number of nearest neighbors.
+     *
+     * @return k the number of nearest neighbors.
+     */
+    public int getNumNeighbors()
+    {
+        return this.k;
+    }
+
+    /**
+     * Gets the number of clusters.
+     *
+     * @return the cluster count.
+     */
+    public int getClusterCount()
+    {
+        return this.clusterCount;
+    }
+
+    /**
+     * Sets the similarity score (mutually shared nearest neighbors) threshold.
+     *
+     * @param eps The minimum similarity score required.
+     */
+    public void setSimilarity(int eps)

```

Comment:
Minor naming: Given this is a minimum threshold maybe call it setMinSimilarity? Or setSimilarityThreshold? (Also change the eps to match the name)